src/main/java/de/unistuttgart/crosswordbackend/service/StatisticService.java

```diff
@@ -0,0 +1,178 @@
+     *
+     * @param problematicQuestions the list of problematic questions where the right answered counter should be increased
+     * @param gameResult the game result which contains the answered questions
+     */
+    private void countRightAnsweredQuestions(
+        final List<ProblematicQuestion> problematicQuestions,
+        final GameResult gameResult
+    ) {
+        gameResult
+            .getAnswers()
+            .parallelStream()
+            .filter(GameAnswer::isCorrect)
+            .forEach(gameAnswer ->
+                problematicQuestions
+                    .stream()
+                    .filter(problematicQuestion ->
+                        problematicQuestion.getQuestion().getQuestionText().equals(gameAnswer.getQuestion())
+                    )
+                    .findAny()
+                    .ifPresent(ProblematicQuestion::addCorrectAnswer)
+            );
+    }
+
+    /**
+     * Counts amount of wrong answered questions and adds them to the problematic question counter
+     *
+     * @param problematicQuestions the list of problematic questions where the wrong answered counter should be increased
+     * @param gameResult the game result which contains the answered questions
+     */
+    private void countWrongAnsweredQuestions(
+        final List<ProblematicQuestion> problematicQuestions,
+        final GameResult gameResult
+    ) {
+        gameResult
+            .getAnswers()
+            .parallelStream()
+            .filter(answer -> !answer.isCorrect())
+            .forEach(gameAnswer ->
+                problematicQuestions
+                    .stream()
+                    .filter(problematicQuestion ->
+                        problematicQuestion.getQuestion().getQuestionText().equals(gameAnswer.getQuestion())
+                    )
+                    .findAny()
+                    .ifPresent(ProblematicQuestion::addWrongAnswer)
+            );
+    }
+
+    /**
+     * Sorts the list of problematic questions by the amount of wrong answers to attempts
+     *
+     * @param problematicQuestions the list of problematic questions to sort
+     */
+    private void sortProblematicQuestionsByPercentageWrongAnswers(
+        final List<ProblematicQuestion> problematicQuestions
+    ) {
+        problematicQuestions.sort((o1, o2) -> {
+            final double percantageWrong1 = (double) o1.getWrongAnswers() / (double) o1.getAttempts();
+            final double percantageWrong2 = (double) o2.getWrongAnswers() / (double) o2.getAttempts();
+            return Double.compare(percantageWrong2, percantageWrong1);
+        });
+    }
+
+    /**
+     * Returns a list of the time spent distribution of a minigame
+     *
+     * @param configurationId the configuration id of the minigame
+     * @return a list of the time spent distribution of a minigame
+     */
+    public List<TimeSpentDistribution> getTimeSpentDistributions(final UUID configurationId) {
```

Comment:
While your hardcoded approach probably works, we probably could have simplified things a lot using a fitting SQL query.
But it is what it is.